kazoo/protocol/connection.py

```diff
@@ -660,13 +686,53 @@ def _connect(self, host, port):
             client._session_callback(KeeperState.CONNECTED_RO)
             self._ro_mode = iter(self._server_pinger())
         else:
-            client._session_callback(KeeperState.CONNECTED)
             self._ro_mode = None
-
-        for scheme, auth in client.auth_data:
-            ap = Auth(0, scheme, auth)
-            zxid = self._invoke(connect_timeout / 1000.0, ap, xid=AUTH_XID)
-            if zxid:
-                client.last_zxid = zxid
+            if client.use_sasl and self.sasl_cli is None:
```

Comment:
@StephenSorriaux I just realized you deleted the blank line and indented the following code so that it's under the `else` in 688... why?

Previously it would run regardless of the code in lines 685-689, but now it's only if you hit the `else` clause...

You have paid attention to details in your PR, so my guess is that I'm just not understanding something, so can you clarify?

Reply:
@jeffwidman You're right, I clearly overlooked this part when adding the behaviour handling the case where pure-sasl's lib is absent. There is no reason not doing auth on RO connection... I'm sorry, I will submit a PR tomorow to fix that